### PR TITLE
use audio cue to signal no inlay hints can be read

### PR DESCRIPTION
**Pull request by jrieken**
This PR extends https://github.com/microsoft/vscode/issues/142532 by playing an audio cue whenever users ask to read the current line with inlay hint but when there are none.

### CODE REVIEW

**Comment by jrieken** [comment]
fyi @hediet and @isidorn - let me know what you think about this audio cue usage and if that's something that we want.

The change is a bit complex because audio cues are workbench only. Therefore I had to extract some inlay hint bits into workbench as well. The gist is

* a new cue for no inlay hints (uses error source)
* play that cue when asking for inlay hints when there are none

File: src/vs/workbench/contrib/audioCues/browser/audioCueService.ts [hunk 1]
```diff
@@ -165,6 +165,12 @@ export class AudioCue {
 		settingsKey: 'audioCues.debuggerStoppedOnBreakpoint',
 	});
 
+	public static readonly noInlayHints = AudioCue.register({
+		name: localize('audioClues.noInlayHints', 'Line has no Inlay Hints'),
```
Comment (hediet) [inline src/vs/workbench/contrib/audioCues/browser/audioCueService.ts:169]: This is being read out as is in the audio cue help.
My first thought was that this is being played for every line that has no audio cue.

I suggest to change the description to "No Inlay Hints available for the current line" to indicate that it is different than the other audio cues.

File: src/vs/workbench/contrib/inlayHints/browser/inlayHintsAccessibilty.ts [hunk 2]
```diff
@@ -47,13 +61,13 @@ export class InlayHintsAccessibility {
...
 
-	reset(): void {
+	private _reset(): void {
 		dom.clearNode(this._ariaElement);
 		this._sessionDispoosables.clear();
 		this._ctxIsReading.reset();
 	}
 
-	async read(line: number, hints: InlayHintItem[]) {
+	private async _(line: number, hints: InlayHintItem[]) {
```
Comment (hediet) [inline src/vs/workbench/contrib/inlayHints/browser/inlayHintsAccessibilty.ts:70]: Huh, is this method name intentional?

**Review by hediet — APPROVED** [review]
Nice ;) Left some feedback.

File: src/vs/workbench/contrib/audioCues/browser/audioCueService.ts (hunk 1 above)
Comment (jrieken) [reply src/vs/workbench/contrib/audioCues/browser/audioCueService.ts:169]: Makes sense

File: src/vs/workbench/contrib/inlayHints/browser/inlayHintsAccessibilty.ts (hunk 2 above)
Comment (jrieken) [reply src/vs/workbench/contrib/inlayHints/browser/inlayHintsAccessibilty.ts:70]: :-)

File: src/vs/workbench/contrib/audioCues/browser/audioCueService.ts [hunk 3]
```diff
@@ -165,6 +165,12 @@ export class AudioCue {
 		settingsKey: 'audioCues.debuggerStoppedOnBreakpoint',
 	});
 
+	public static readonly noInlayHints = AudioCue.register({
+		name: localize('audioClues.noInlayHints', 'No Inlay Hints available for the current line'),
+		sound: Sound.error,
+		settingsKey: 'audioClues.noInlayHints'
```
Comment (isidorn) [inline src/vs/workbench/contrib/audioCues/browser/audioCueService.ts:171]: It is not `audioClues` it is `audioCues` :)

Comment (jrieken) [reply src/vs/workbench/contrib/audioCues/browser/audioCueService.ts:171]: fixed in main

**Comment by isidorn** [comment]
Cool work!
I think this should follow the pattern that we do for all other features. If the audioCue setting is "on" that means the sound is played when the cursor is on a line with inlay hints. So a user can trigger an additional command to learn more.
I do not think we need a sound for "no inlay hints". Silence will carry that meaning.

**Comment by isidorn** [comment]
I tried it out and some more feedback:

1. "Read line with inlay hints" -> I think we should rename this command to be be more general and that we can potentially use it for all editor richness in the future. I am not sure what is the right name, but I will think about it
2. I will ping the screen reader users so they try it out. @jrieken do you know if inlay hints are available in Java, Python, C++ just so I know how to tell users to try it out
3. I am a bit worried about only `esc` and `tab` going out of this mode. Ideally any key or at least left/right/up/down would take you out
4. I do not think there is a need for "stop reading" to be in the command palette

All in all, cool work and I think we should just re-think some things so we can build more on top of this :)

**Pull request merged**